src/main/scala/beam/agentsim/infrastructure/IndexerForZonalParkingManager.scala

```diff
@@ -70,9 +70,8 @@ class IndexerForZonalParkingManager(resources: Map[StallAttributes, StallValues]
 
     tazWithDistance.view
       .map { case (taz, _) => find(key(taz.tazId, reservedFor)) }
-      .filter(_.isDefined)
-      .headOption
-      .getOrElse(None)
+      .find(_.isDefined)
```

Comment:
This can be cleaned even further:

    .collectFirst{ case Some(x) => x }

Reply:
collectFirst will extract using destructor and then instantiate again a new `Option/Some` (it was already option previously). Syntax is a little bit better, but not sure it is worthy for this specific case.